curator-recipes/src/main/java/org/apache/curator/framework/recipes/locks/ChildReaper.java

```diff
@@ -173,32 +191,40 @@ public boolean removePath(String path)
         return paths.remove(PathUtils.validatePath(path));
     }
 
-    private static ScheduledExecutorService newExecutorService()
+    public static ScheduledExecutorService newExecutorService()
     {
         return ThreadUtils.newFixedThreadScheduledPool(2, "ChildReaper");
     }
 
     private void doWork()
     {
-        for ( String path : paths )
+        if (shouldDoWork())
```

Comment:
That race condition is already present with the leader election in the Reaper itself as it could lose leadership after scheduling a check for some time in the future, so this is at least no worse than the current state of things.  And the race condition doesn't matter too much anyway as the reapers don't use leader election for correctness, but rather because otherwise you get a bunch of logspam about trying to delete already deleted nodes.